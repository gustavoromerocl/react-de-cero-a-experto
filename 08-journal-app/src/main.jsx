import React from 'react'
import ReactDOM from 'react-dom/client'
import './index.css'
import { JournalApp } from './JournalApp.jsx'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <JournalApp />
  </React.StrictMode>,
)
