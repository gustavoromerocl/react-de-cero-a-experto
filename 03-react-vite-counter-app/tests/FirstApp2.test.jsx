import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('debe validar el titulo', () => {
  const title = 'Hola soy gustavo';
  test('debe hacer match con el snapshot', () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  })

  test('debe hacer match con el snapshot', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByText(title).toBeTruthy());
  })

  test('Debe validar el tÃ­tulo en un h1', () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole('heading', { level: 1 }).innerHtml).toContain(title);
  })

  test('Debe de mostrar el subtitulo enviado por props', () => {
    const subTitle = 'hello world';
    render(<FirstApp
      title={title}
      subTitle={subTitle}
    />);
    expect(screen.getAllByText(subTitle).length).toBe(2);
  })
})