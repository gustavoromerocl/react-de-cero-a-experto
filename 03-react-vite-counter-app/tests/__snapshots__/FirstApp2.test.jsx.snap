// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debe validar el titulo debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola soy gustavo
     
  </h1>
  <p>
    No hay subtítulo
  </p>
  <p>
    No hay subtítulo
  </p>
  <p>
    Fernando Herrera
  </p>
</div>
`;
